src/reanimated2/NativeMethods.ts

```diff
@@ -219,3 +223,47 @@ if (IS_NATIVE) {
     );
   };
 }
+
+export let setNativeProps: <T extends Component>(
+  animatedRef: AnimatedRef<T>,
+  updates: StyleProps
+) => void;
+
+if (isWeb()) {
+  setNativeProps = (_animatedRef, _updates) => {
+    throw new Error(
+      '[Reanimated] setNativeProps() is not supported on web yet.'
```

Comment:
Currently, there are two technical difficulties that prevent us from implementing this properly:
1. In Reanimated, there's no single pipeline for updating props on web. We have multiple functions (`updateProps`, `_updatePropsJS`, `updatePropsDOM`, `setNativeProps`) that perform additional operations (like diffing with old props, converting transforms etc.). We'll need to centralize that.
2. Animated refs return host instance, i.e. `viewTag` or `HTMLElement`. We'll need to refactor `useAnimatedRef` so that it returns the animated component as React component.

I added basic web support in 84f1cd2fde9be93e07589e4e6b975c7baf1258f8 and fixed color handling on native in cf2318db9c1a85002189d0a842eeedc470ed7826. When `setNativeProps` is used in JSReanimated mode on native, a proper warning will appear.